shipyard_blueprints/magnite/shipyard_magnite/cli/update_campaign_budget.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+import sys
+
+from shipyard_templates import ShipyardLogger, ExitCodeException
+from shipyard_templates.errors import EXIT_CODE_UNKNOWN_ERROR
+from shipyard_magnite import MagniteClient, utils
+from shipyard_magnite.errs import (
+    EXIT_CODE_UPDATE_ERROR,
+    EXIT_CODE_FILE_NOT_FOUND,
+    EXIT_CODE_PARTIAL_FAILURE,
+)
+from shipyard_magnite.dataclasses.targeting_spend_profile import TargetingSpendProfile
+
+
+logger = ShipyardLogger.get_logger()
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        description="Update campaign budgets using Magnite API"
+    )
+    parser.add_argument("--username", required=True, help="Magnite API username")
+    parser.add_argument("--password", required=True, help="Magnite API password")
+    parser.add_argument(
+        "--file", required=True, help="Path to the CSV file containing budget data"
+    )
+    return parser.parse_args()
+
+
+def process_campaign(client, campaign_id, budget_data):
+    """
+    Process a single campaign's budget.
+    """
+    try:
+        logger.info(f"Processing campaign ID: {campaign_id}...")
+
+        budgets = utils.transform_data_to_budgets(budget_data)
+        valid_budgets = budgets.validate()
+        report = budgets.report_validation_results()
+
+        if not valid_budgets.items:
+            logger.error(f"No valid budgets found for campaign ID: {campaign_id}")
+            return {"errors": [campaign_id], "report": {campaign_id: report}}
+
+        if len(valid_budgets.items) < len(budgets.items):
+            logger.warning(
+                f"Found invalid budgets for campaign ID: {campaign_id}. "
+                f"Proceeding with the following valid budgets:\n"
+                f"{chr(10).join(str(budget) for budget in valid_budgets.items)}"
+            )
+
+        profile_id = client.get_campaign_by_id(campaign_id)["targeting_spend_profile"][
+            "id"
+        ]
+        spend_profile = TargetingSpendProfile(id=profile_id, budgets=valid_budgets)
+        client.update_campaign_budgets(campaign_id, spend_profile)
+
+        if len(valid_budgets.items) == len(budgets.items):
+            logger.info(f"Successfully updated budgets for campaign ID: {campaign_id}")
+        else:
+            logger.warning(
+                f"Partial success: Updated budgets for campaign ID: {campaign_id}. \n"
+                "Note: Some budget items were invalid and excluded from the update."
+            )
```

Comment:
Is this necessary here given the `<` check above?

Reply:
yes and no.

these logs happen after `client.update_campaign_budgets(campaign_id, spend_profile)` step so in theory something wrong can happen there. 

If I just had it as `Success` felt like it could be misleading to the end user that the update went off without issue. I wanted to make it very clear that yes the edit to the campaign did go through but not all of the budgets that were inputted made its way.